src/sentry/integrations/repositories.py

```diff
@@ -2,19 +2,48 @@
 
 from sentry.constants import ObjectStatus
 from sentry.models import Repository
+from sentry.shared_integrations.exceptions import ApiError
 
 
 class RepositoryMixin(object):
     # whether or not integration has the ability to search through Repositories
     # dynamically given a search query
     repo_search = False
 
-    def get_stacktrace_link(self, repo, path, version):
+    def format_source_url(self, repo_name, filepath, branch):
+        """
+        Formats the source code url used for stack trace linking.
+        """
+        raise NotImplementedError
+
+    def check_file(self, repo_name, filepath, branch):
```

Comment:
@MeredithAnya I think it might make more sense for this to be `get_source_url` and call `format_source_url` so we don't have to call `format_source_url` each time we call `check_file`.